src/cdk/table/table.ts

```diff
@@ -535,7 +535,7 @@ export class CdkTable<T> implements CollectionViewer, OnInit, AfterContentChecke
    */
   private _getHeaderCellTemplatesForRow(headerDef: CdkHeaderRowDef): CdkHeaderCellDef[] {
     if (!headerDef || !headerDef.columns) { return []; }
-    return headerDef.columns.map(columnId => {
+    return Array.from(headerDef.columns).map(columnId => {
```

Comment:
Fun fact: the second parameter of `Array.from` is a `map` function so you don't have to do `.map` afterwards.

Reply:
Today I learned.  Updated.